### ISSUE TITLE
debug ：TypeError: tuple indices must be integers or slices, not tuple

### ISSUE THREAD

**Issue by YHQCP3** [dyhBUPT/StrongSORT#122]
When I run the code, I encounter an error. How should I fix this? Specifically, the error occurs when I run the following command:
(3.9.7) F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master>python strong_sort.py MOT20 test --BoT --NSA --EMA --MC --woC --AFLink --GSI

The output and error message are as follows:
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.1.1\plugins\python-ce\helpers\pydev\pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.1.1\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\strong_sort.py", line 24, in <module>
    run(
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\deep_sort_app.py", line 213, in run
    visualizer.run(frame_callback)
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\application_util\visualization.py", line 80, in run
    frame_callback(self, self.frame_idx)
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\deep_sort_app.py", line 190, in frame_callback
    tracker.update(detections)
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\deep_sort\tracker.py", line 70, in update
    self._match(detections)
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\deep_sort\tracker.py", line 126, in _match
    linear_assignment.min_cost_matching(
  File "F:\xiazai1\ultralytics-main\strongsort\StrongSORT-master\deep_sort\linear_assignment.py", line 66, in min_cost_matching
    if col not in indices[:, 1]:
TypeError: tuple indices must be integers or slices, not tuple

**Comment by YHQCP3** [comment]
I hope that experts can provide an answer!  
Thank you!！！

**Comment by BrendanWooAdmetal** [comment]
I encountered this issue as well, as a result of imports.

This import is deprecated.
`from sklearn.utils.linear_assignment_ import linear_assignment`

So I used this import instead.
`from scipy.optimize import linear_sum_assignment as linear_assignment`

However, the return type of the new import is different, returning a tuple of numpy arrays instead of a numpy array.
Altering the return of `linear_assignment()` to be in the original format resolved this issue for me.

This can be done in `linear_assignment.py, line 61` by adding the two lower lines.

```
    indices = np.array(linear_assignment(cost_matrix_))
    indices = np.asarray(indices)
    indices = np.transpose(indices)
```

This solution was found at this link:
https://stackoverflow.com/questions/57369848/how-do-i-resolve-use-scipy-optimize-linear-sum-assignment-instead

I'm unsure if this is the exact same issue you are experiencing, and I see this issue is closed, but hopefully this will help any future developers that encounter the same issue.
